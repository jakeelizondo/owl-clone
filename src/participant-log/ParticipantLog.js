import React from 'react';
import Participant from '../participant/Participant.js';

function ParticipantLog(props) {
  let participantsSorted = props.participants.sort(function (a, b) {
    return b.onStage - a.onStage;
  });

  let participantComps = participantsSorted.map((participant) => {
    return (
      <Participant
        key={participant.id}
        avatar={participant.avatar}
        id={participant.id}
        inSession={participant.inSession}
        name={participant.name}
        onStage={participant.onStage}
      />
    );
  });

  console.log(participantComps);

  return <div>{participantComps}</div>;
}

export default ParticipantLog;
