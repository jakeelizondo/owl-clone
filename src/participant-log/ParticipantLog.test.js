import React from 'react';
import ReactDOM from 'react-dom';
import ParticipantLog from './ParticipantLog';
import renderer from 'react-test-renderer';
import participants from '../participants';

describe('ParticipantLog Component', () => {
  // smoke test for app component

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ParticipantLog participants={participants} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<ParticipantLog key={1} participants={participants} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
