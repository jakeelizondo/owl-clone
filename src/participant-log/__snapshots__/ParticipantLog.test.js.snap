// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipantLog Component renders the UI as expected 1`] = `
<div>
  <div
    className="participant"
  >
    <div>
      <img
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
      <h2>
        Koren Templeton
      </h2>
      <p>
        on stage
      </p>
    </div>
  </div>
  <div
    className="participant"
  >
    <div>
      <img
        src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
      />
      <h2>
        Ashla Attwool
      </h2>
      <p>
        on stage
      </p>
    </div>
  </div>
  <div
    className="participant"
  >
    <div>
      <img
        src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
      />
      <h2>
        Caty Flucker
      </h2>
      <p />
    </div>
  </div>
  <div
    className="participant"
  >
    <div>
      <img
        src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
      />
      <h2>
        Axe Kubicka
      </h2>
      <p />
    </div>
  </div>
  <div
    className="participant"
  >
    <div>
      <img
        src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
      />
      <h2>
        Frank Runciman
      </h2>
      <p />
    </div>
  </div>
</div>
`;
